src/BigQuery/BigQueryClient.php

```diff
@@ -0,0 +1,355 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\BigQuery;
+
+use Google\Cloud\BigQuery\Connection\ConnectionInterface;
+use Google\Cloud\BigQuery\Connection\Rest;
+
+/**
+ * Google Cloud BigQuery client. Allows you to create, manage, share and query
+ * data. Find more information at
+ * [Google Cloud BigQuery Docs](https://cloud.google.com/bigquery/what-is-bigquery).
+ */
+class BigQueryClient
+{
+    use JobConfigurationTrait;
+
+    const SCOPE = 'https://www.googleapis.com/auth/bigquery';
+    const INSERT_SCOPE = 'https://www.googleapis.com/auth/bigquery.insertdata';
+
+    /**
+     * @var ConnectionInterface $connection Represents a connection to BigQuery.
+     */
+    protected $connection;
+
+    /**
+     * @var string The project ID created in the Google Developers Console.
+     */
+    private $projectId;
+
+    /**
+     * Create a BigQuery client.
+     *
+     * Example:
+     * ```
+     * use Google\Cloud\BigQuery\BigQueryClient;
+     *
+     * $bigQuery = new BigQueryClient([
+     *     'projectId' => 'myAwesomeProject'
+     * ]);
+     * ```
+     *
+     * @param array $config {
+     *     Configuration options.
+     *
+     *     @type string $projectId The project ID from the Google Developer's
+     *           Console.
+     *     @type callable $authHttpHandler A handler used to deliver Psr7
+     *           requests specifically for authentication.
+     *     @type callable $httpHandler A handler used to deliver Psr7 requests.
+     *     @type string $keyFile The contents of the service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type int $retries Number of retries for a failed request. Defaults
+     *           to 3.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     */
+    public function __construct(array $config = [])
+    {
+        if (!isset($config['projectId'])) {
+            throw new \InvalidArgumentException('A projectId is required.');
+        }
+
+        if (!isset($config['scopes'])) {
+            $config['scopes'] = [self::SCOPE];
+        }
+
+        $this->connection = new Rest($config);
+        $this->projectId = $config['projectId'];
+    }
+
+    /**
+     * Runs a BigQuery SQL query in a synchronous fashion. Rows are returned
+     * immediately as long as the query completes within a specified timeout.
```

Comment:
Very good question. I will add some more documentation around how to handle this situation.